### PR TITLE
Fix reading JSON from pipe

### LINKED ISSUE #1: Error when piping JSON to jpterm
Thanks @springcomp for forking and updating this project :slightly_smiling_face:.

I have found that piping into `jpterm` generates an error:

```bash
cat defaults/main.yml | yq -o=json | jpterm 
Traceback (most recent call last):
  File "/home/chris/.local/bin/jpterm", line 6, in <module>
    jpterm.main()
  File "/home/chris/.local/pipx/venvs/jmespath-community-terminal/lib/python3.11/site-packages/jpterm.py", line 231, in main
    display.main(screen=screen)
  File "/home/chris/.local/pipx/venvs/jmespath-community-terminal/lib/python3.11/site-packages/jpterm.py", line 148, in main
    self.loop.run()
  File "/home/chris/.local/pipx/venvs/jmespath-community-terminal/lib/python3.11/site-packages/urwid/main_loop.py", line 287, in run
    self._run()
  File "/home/chris/.local/pipx/venvs/jmespath-community-terminal/lib/python3.11/site-packages/urwid/main_loop.py", line 385, in _run
    self.event_loop.run()
  File "/home/chris/.local/pipx/venvs/jmespath-community-terminal/lib/python3.11/site-packages/urwid/main_loop.py", line 790, in run
    self._loop()
  File "/home/chris/.local/pipx/venvs/jmespath-community-terminal/lib/python3.11/site-packages/urwid/main_loop.py", line 827, in _loop
    self._watch_files[fd]()
  File "/home/chris/.local/pipx/venvs/jmespath-community-terminal/lib/python3.11/site-packages/urwid/raw_display.py", line 417, in <lambda>
    event_loop, callback, self.get_available_raw_input())
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/chris/.local/pipx/venvs/jmespath-community-terminal/lib/python3.11/site-packages/urwid/raw_display.py", line 449, in get_available_raw_input
    codes = self._get_gpm_codes() + self._get_keyboard_codes()
                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/chris/.local/pipx/venvs/jmespath-community-terminal/lib/python3.11/site-packages/urwid/raw_display.py", line 523, in _get_keyboard_codes
    code = self._getch_nodelay()
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/chris/.local/pipx/venvs/jmespath-community-terminal/lib/python3.11/site-packages/urwid/raw_display.py", line 661, in _getch_nodelay
    return self._getch(0)
           ^^^^^^^^^^^^^^
  File "/home/chris/.local/pipx/venvs/jmespath-community-terminal/lib/python3.11/site-packages/urwid/raw_display.py", line 571, in _getch
    return ord(os.read(fd, 1))
           ^^^^^^^^^^^^^^^^^^^
TypeError: ord() expected a character, but string of length 0 found
```

But writing a file and opening it work fine:

```bash
cat defaults/main.yml | yq -o=json > ~/tmp/ansible_defaults.json
jpterm ~/tmp/ansible_defaults.json
```

This isn't a big or urgent problem!

This is on Debian Bookworm with a `pipx` installed version:

```bash
which jpterm
/home/chris/.local/bin/jpterm
/home/chris/.local/bin/jpterm --version
jmespath-term 1.1.1
/usr/bin/python3 --version
Python 3.11.2
```

### PR DESCRIPTION
**Pull request by kit494way**
Close #1

I fixed the issue #1 with reference [here](https://github.com/python/cpython/issues/36029#issuecomment-1093968541).
I checked that this change works in Python 3.11.4, 3.10.12, 3.9.17 and 3.8.17.

### CODE REVIEW

**Comment by kit494way** [comment]
@springcomp
Is there anything I can do to get this and other [pull requests](https://github.com/jmespath-community/jmespath.terminal/pulls?q=is%3Apr+is%3Aopen) reviewed and merged?

**Review by springcomp — COMMENTED** [review]
Thanks a lot for your contribution.

Would you please abstract away this code into separate functions?

File: jpterm.py [hunk 1]
```diff
@@ -187,8 +188,21 @@ def _load_input_json(filename):
     elif not os.isatty(sys.stdin.fileno()):
         # If stdin is a pipe, we need read the JSON from
         # stdin and then reset stdin this back to the controlling tty.
-        input_json = json.loads(sys.stdin.read())
-        sys.stdin = open(os.ctermid(), 'r')
+        # Replace fd(0) with tty instead of modifying sys.stdin.
+        # See https://github.com/python/cpython/issues/36029#issuecomment-1093968541 # noqa
+        buf = io.BytesIO()
```
Comment (springcomp) [inline jpterm.py:193]: That would be great to split this code into a separate function or better a separate object ?

Comment (kit494way) [reply jpterm.py]: @springcomp 
Thanks for your review. I split the code into a separate function.

**Review by springcomp — APPROVED** [review]
👍

**Pull request merged**